### ISSUE TITLE
adas

### ISSUE THREAD

**Issue by uthambathoju** [utham369/CoronaryHeartDisease#1]
data_tofit = [1.6800483    -1.641695388; 
              0.501309281  -0.977697538; 
              1.528012113   0.52771122;
              1.70012253    1.711524991; 
              1.992493625   1.891000015;
              2.706075824  -0.463427794;
              2.994931927  -0.443566619;
              3.491852811  -1.275179133;
              3.501191722  -0.690499597;
              4.459924502  -5.516130799;
              4.936965851  -6.001703074;
              5.023289852  -8.36416901;
              5.04233698   -7.924477517;
              5.50739285  -10.77482371;
              5.568665171 -10.9171878]

**Comment by utham369** [comment]
x, y = data_tofit[:,1], data_tofit[:,2]

function parabfit(x)
    return a * x^2 + b * x + c
end


plot(parabfit, -5, 5)

a, b, c = -1, 3, 0.8
plot(x, y, linetype=:scatter)
plot!(parabfit, 0, 6)



a, b, c = -1, 3, 0.8
plot(x, y, linetype=:scatter)
plot!(parabfit, 0, 15)

**Comment by utham369** [comment]
a, b, c = -1, 4, -3
plot(x, y, linetype=:scatter)
plot!(parabfit, 0, 6)

plot(x, y, linetype=:scatter)

**Comment by utham369** [comment]
NLP Engine(Chart abstraction output) generates diagnoses conditions from medical chart (EMR). These conditions can be false/incorrectly tagged as an output due to various reasons like below:
OCR accuracy
NLP challenges such as word boundaries, context understanding  etc.
Conditions present in cover pages e.g.. PAF pages

The current NLP engine has approx. 70% (All Historical Data) precision rate.


Designed a machine learning solution to analyze multiple data sources, understand the impact to proactively reduce the false positive HCCs in NLP outcomes based on NLP Member/ Winner  data.
Leveraged historical data to model and understand diagnosis conditions impact on false positive HCCs.
Applied best practices in model selection and testing to build an effective False positive HCCs detection model in medical charts(EMR).
By using the ML model post NLP engine, the percentage of False conditions reduced to 5% without significant impact on Recall.
 NLP Engine Precision has been improved from ~70%  to ~95%
.


Finding an exact HCCs in a chart  is always important for the     acceptance by CMS. This solution also acts as an underlying tool for other commercial features in NLP Clinical Data Program..
QRA Evaluation : Higher submission Success rate for suspects match
Chart Prioritizing : Higher NLP precision will have a better NLP Chart Value Score.
 Chart Preparation : Human coder has more confidence of chart preparation with bookmarking.
Operational Cost reduction : In Second Level Review , Fewer False HCC submission reduces operational cost and saves time

**Comment by utham369** [comment]
you then hope that this leads to also doing well on the dev set. And you then hope that this also does well on the test set. And finally, you hope that doing well on the test set on the cost function results in your system performing in the real world. So you hope that this resolves in happy cat picture app users, for example. So to relate back to the TV tuning example, if the picture of your TV was either too wide or too narrow, you wanted one knob to tune in order to adjust that. You don't want to have to carefully adjust five different knobs, which also affect different things. You want one knob to just affect the width of your TV image. So in a similar way, if your algorithm is not fitting the training set well on the cost function, you want one knob, yes, that's my attempt to draw a knob. Or maybe one specific set of knobs that you can use, to make sure you can tune your algorithm to make it fit well on the training set. So the knobs you use to tune this are, you might train a bigger network.
Play video starting at :6:16 and follow transcript6:16
Or you might switch to a better optimization algorithm, like the Adam optimization algorithm, and so on, into some other options we'll discuss later this week and next week.
Play video starting at :6:28 and follow transcript6:28
In contrast, if you find that the algorithm is not fitting the dev set well, then there's a separate set of knobs. Yes, that's my not very artistic rendering of another knob, you want to have a distinct set of knobs to try. So for example, if your algorithm is not doing well on the dev set, it's doing well on the training set but not on the dev set, then you have a set of knobs around regularization that you can use to try to make it satisfy the second criteria. So the analogy is, now that you've tuned the width of your TV set, if the height of the image isn't quite right, then you want a different knob in order to tune the height of the TV image. And you want to do this hopefully without affecting the width of your TV image too much. And getting a bigger training set would be another knob you could use, that helps your learning algorithm generalize better to the dev set. Now, having adjusted the width and height of your TV image, well, what if it doesn't meet the third criteria? What if you do well on the dev set but not on the test set? If that happens, then the knob you tune is, you probably want to get a bigger dev set. Because if it does well on the dev set but not the test set, it probably means you've overtuned to your dev set, and you need to go back and find a bigger dev set.
Play video starting at :7:52 and follow transcript7:52
And finally, if it does well on the test set, but it isn't delivering to you a happy cat picture app user, then what that means is that you want to go back and change either the dev set or the cost function.
Play video starting at :8:13 and follow transcript8:13
Because if doing well on the test set according to some cost function doesn't correspond to your algorithm doing what you need it to do in the real world, then it means that either your dev test set distribution isn't set correctly, or your cost function isn't measuring the right thing. I know I'm going over these examples quite quickly, but we'll go much more into detail on these specific knobs later this week and next week. So if you aren't following all the details right now, don't worry about it. But I want to give you a sense of this orthogonalization process,

**Comment by utham369** [comment]
For a supervised learning system to do well, you usually need to tune the knobs of your system to make sure that four things hold true. First, is that you usually have to make sure that you're at least doing well on the training set. So performance on the training set needs to pass some acceptability assessment. For some applications, this might mean doing comparably to human level performance. But this will depend on your application, and we'll talk more about comparing to human level performance next week.

**Comment by utham369** [comment]
But after doing well on the training sets, you then hope that this leads to also doing well on the dev set. And you then hope that this also does well on the test set. And finally, you hope that doing well on the test set on the cost function results in your system performing in the real world.

**Comment by utham369** [comment]
![image](https://user-images.githubusercontent.com/16901759/113087138-9f961300-9200-11eb-964e-d3af0c69aefe.png)

**Comment by uthambathoju** [comment]
To build a machine learning solution to analyze multiple data sources, understand the impact to proactively reduce the false positives in NLP outcomes.
Designed a predictive and stable model based on NLP Member/ Winner  data.
Leveraged historical data to model and understand diagnosis conditions impact on false positive.
Applied best practices in model selection and testing to build an effective False Positive detection model to analyze the medical charts.
Modeling: Used a different classification and regression models.

**Comment by uthambathoju** [comment]
import requests
from bs4 import BeautifulSoup
import numpy as np
import pandas as pd
import time
page_url ='https://www.mwnation.com/section/chichewa/page/'

urls = ['https://www.mwnation.com/section/chichewa/']
for i in range(2, 77):
    urls.append(page_url + str(i) +'/')
   

import requests
from bs4 import BeautifulSoup
ignore_list = ['section','about-npl', 'imagination', 'adverts', 'rate-card', 'contact-us','wp-content',
               'court-snubs-mcp-in-commissioners-case','values','our-philosophy','editorial-policy',
               'advertising-policy','code-of-conduct','plagiarism-disclaimer','disclaimer','privacy-policy','terms-of-use',
               'lessons-from-derek-chauvin-case','careening-dangerously-like-kabaza-motorcyclist','progress-needs-positive-reporting',
               'fighting-misinformation-with-information','wp-contentuploads202103WFP-Afikepo-1-2.pdf','minister-sues-2-over-social-media-post','malawi-hunts-for-forex-in-ghana',
               'chakweras-mining-dream','mcp-to-elect-vice-president-soon','court-snubs-mcp-in-commissioners-case','values'] 

#url = 'https://www.mwnation.com/section/chichewa/page/2/'
urls_data  = []

for url in urls:
    reqs = requests.get(url)
    soup = BeautifulSoup(reqs.text, 'html.parser')
    for link in soup.find_all('a'):
        flag = True
        if "https://www.mwnation.com/" in link.get('href'):
            value = link.get('href')
            for keyword in ignore_list:
                if value.find(keyword) != -1:
                    flag = False

            if flag:
                urls_data.append(value)
urls_data = list(set(urls_data))
urls_data.remove('https://www.mwnation.com/')
print(len(urls_data))


idx = 0
data_dict = {}
paragraph = []
urls_data = ['https://www.mwnation.com/zokolola-zichuluka/', 'https://www.mwnation.com/boma-ligulitsa-chimanga-pamtengo-wa-sabuside/']
for url in urls_data:
    r1 = requests.get(url)
    print("0")
    soup = BeautifulSoup(r1, 'html.parser')
    print("1")
    for link in soup.find_all('p'):
        if 'Nation Publications Limited' not in link.text:
            print("2")
            title = soup.title.text
            text_data = link.text.strip()
            data_dict[idx] = paragraph.append((title, paragraph))
            print("3")
            idx += 1
            print(idx)

**Comment by utham369** [comment]
`import requests
from bs4 import BeautifulSoup
import numpy as np
import pandas as pd
import time
page_url ='https://www.mwnation.com/section/chichewa/page/'

urls = ['https://www.mwnation.com/section/chichewa/']
for i in range(2, 77):
    urls.append(page_url + str(i) +'/')

ignore_list = ['section','about-npl', 'imagination', 'adverts', 'rate-card', 'contact-us','wp-content',
'court-snubs-mcp-in-commissioners-case','values','our-philosophy','editorial-policy',
'advertising-policy','code-of-conduct','plagiarism-disclaimer','disclaimer','privacy-policy','terms-of-use',
'lessons-from-derek-chauvin-case','careening-dangerously-like-kabaza-motorcyclist','progress-needs-positive-reporting',
'fighting-misinformation-with-information','wp-contentuploads202103WFP-Afikepo-1-2.pdf','minister-sues-2-over-social-media-post','malawi-hunts-for-forex-in-ghana',
'chakweras-mining-dream','mcp-to-elect-vice-president-soon','court-snubs-mcp-in-commissioners-case','values']

#url = 'https://www.mwnation.com/section/chichewa/page/2/'
urls_data = []

for url in urls:
    reqs = requests.get(url)
    soup = BeautifulSoup(reqs.text, 'html.parser')
    for link in soup.find_all('a'):
        flag = True
        if "https://www.mwnation.com/" in link.get('href'):
            value = link.get('href')
            for keyword in ignore_list:
                if value.find(keyword) != -1:
                    flag = False

            if flag:
                urls_data.append(value)
urls_data = list(set(urls_data))
urls_data.remove('https://www.mwnation.com/')
print(len(urls_data))

idx = 0
data_dict = {}
paragraph = []
urls_data = ['https://www.mwnation.com/zokolola-zichuluka/', 'https://www.mwnation.com/boma-ligulitsa-chimanga-pamtengo-wa-sabuside/']
for url in urls_data:
    r1 = requests.get(url)
    print("0")
    soup = BeautifulSoup(r1, 'html.parser')
    print("1")
    for link in soup.find_all('p'):
        if 'Nation Publications Limited' not in link.text:
            print("2")
            title = soup.title.text
            text_data = link.text.strip()
            data_dict[idx] = paragraph.append((title, paragraph))
            print("3")
            idx += 1
            print(idx)`

**Comment by uthambathoju** [comment]
`
import requests
from bs4 import BeautifulSoup
import numpy as np
import pandas as pd
import time
page_url ='https://www.mwnation.com/section/chichewa/page/'

urls = ['https://www.mwnation.com/section/chichewa/']
for i in range(2, 77):
    urls.append(page_url + str(i) +'/')

ignore_list = ['section','about-npl', 'imagination', 'adverts', 'rate-card', 'contact-us','wp-content',
'court-snubs-mcp-in-commissioners-case','values','our-philosophy','editorial-policy',
'advertising-policy','code-of-conduct','plagiarism-disclaimer','disclaimer','privacy-policy','terms-of-use',
'lessons-from-derek-chauvin-case','careening-dangerously-like-kabaza-motorcyclist','progress-needs-positive-reporting',
'fighting-misinformation-with-information','wp-contentuploads202103WFP-Afikepo-1-2.pdf','minister-sues-2-over-social-media-post','malawi-hunts-for-forex-in-ghana',
'chakweras-mining-dream','mcp-to-elect-vice-president-soon','court-snubs-mcp-in-commissioners-case','values']

#url = 'https://www.mwnation.com/section/chichewa/page/2/'
urls_data = []

for url in urls:
    reqs = requests.get(url)
    soup = BeautifulSoup(reqs.text, 'html.parser')
    for link in soup.find_all('a'):
        flag = True
        if "https://www.mwnation.com/" in link.get('href'):
            value = link.get('href')
            for keyword in ignore_list:
                if value.find(keyword) != -1:
                    flag = False

            if flag:
                urls_data.append(value)
urls_data = list(set(urls_data))
urls_data.remove('https://www.mwnation.com/')
print(len(urls_data))

idx = 0
data_dict = {}
paragraph = []
urls_data = ['https://www.mwnation.com/zokolola-zichuluka/', 'https://www.mwnation.com/boma-ligulitsa-chimanga-pamtengo-wa-sabuside/']
for url in urls_data:
    r1 = requests.get(url)
    print("0")
    soup = BeautifulSoup(r1, 'html.parser')
    print("1")
    for link in soup.find_all('p'):
        if 'Nation Publications Limited' not in link.text:
            print("2")
            title = soup.title.text
            text_data = link.text.strip()
            data_dict[idx] = paragraph.append((title, paragraph))
            print("3")
            idx += 1
            print(idx) 
`
